src/libraries/System.Collections.Immutable/src/System/Collections/Immutable/ImmutableArray_1.cs

```diff
@@ -883,73 +881,17 @@ public ImmutableArray<T> RemoveRange(ReadOnlySpan<T> items, IEqualityComparer<T>
                 return self.Remove(items[0], equalityComparer);
             }
 
-#nullable disable
-            var itemsMultiSet = new Dictionary<T, int>(equalityComparer);
-#nullable restore
-            int nullValueCount = 0;
-            foreach (ref readonly T item in items)
-            {
-                if (item == null)
-                {
-                    nullValueCount++;
-                }
-                else
-                {
-#if NET6_0_OR_GREATER
-#nullable disable
-                    ref int count = ref CollectionsMarshal.GetValueRefOrAddDefault(itemsMultiSet, item, out _);
-#nullable restore
-                    count++;
-#else
-                    if (itemsMultiSet.TryGetValue(item, out int count))
-                    {
-                        itemsMultiSet[item] = count + 1;
-                    }
-                    else
-                    {
-                        itemsMultiSet[item] = 1;
-                    }
-#endif
-                }
-            }
-
-            List<int>? indicesToRemove = null;
-            T[] selfArray = self.array!;
-            for (int i = 0; i < selfArray.Length; i++)
+            var indicesToRemove = new SortedSet<int>();
+            foreach (var item in items)
```

Comment:
Please use explicit type names instead of var here.

Reply:
Fixed.